openwebstart/src/main/java/com/openwebstart/proxy/pac/PacFileEvaluator.java

```diff
@@ -116,6 +113,27 @@ String getProxies(final URI uri) {
         if (cachedResult != null) {
             return cachedResult;
         }
+        
+        // pacUrl.openStream() must not be called in constructor, otherwise it will initialize HttpURLConnection.userAgent
+        // before System.setProperty(HTTP_AGENT, httpAgent) is called in net.sourceforge.jnlp.JNLPFile
+        if (pacContents == null) {
+           LOG.debug("Open PAC url '{}'", pacUrl);
+           // need to temporarly deactivate ProxySelector, otherwise pacUrl.openStream() will call getProxies and throw StackOverflowError after a while
+           ProxySelector proxySelector = ProxySelector.getDefault();
+           ProxySelector.setDefault(null);
```

Comment:
I am still hesitating with this line as there are many things running in parallel. I am a little bit afraid that in some cases a connection my slip by before the proxyselector is re-set in the finally. This would lead to nasty race conditions which are very hard to detect and debug...

Would it be possible to compare the `uri` with the `pacUrl` and in case they are the same return `PacConstants.DIRECT`?
If so this could be done as the first thing in the method and thus never proxying the pac url.
As a result it would not be necessary to temporarily disable the proxy.

Also from a readability point of view it would be favorable to extract the loading of the pacContent into a method which returns the pacContent string

```
if (pacContents == null) {
    pacContents = loadPacContent();
}
```